.github/workflows/test.yml

```diff
@@ -66,14 +58,34 @@ jobs:
         uses: actions/cache@v2
         with:
           path: .tox
-          key: tox-cache-${{ matrix.python-version }}-${{ matrix.package }}-${{ matrix.os }}-${{ hashFiles('tox.ini', 'dev-requirements.txt') }}-core
+          key: tox-cache-${{ env.RUN_MATRIX_COMBINATION }}-${{ hashFiles('tox.ini', 'dev-requirements.txt') }}-core
       - name: run tox
-        run: tox -f ${{ matrix.python-version }}-${{ matrix.package }}
+        run: tox -f ${{ matrix.python-version }}-${{ matrix.package }} -- --benchmark-json=${{ env.RUN_MATRIX_COMBINATION }}-benchmark.json
+      - name: Find and merge benchmarks
+        # TODO: Add at least one benchmark to every package type to remove this
+        if: matrix.package == 'core'
+        run: jq -s '.[0].benchmarks = ([.[].benchmarks] | add) | if .[0].benchmarks == null then null else .[0] end' opentelemetry-*/tests/*${{ matrix.package }}*.json > output.json
+      - name: Report on benchmark results
+        # TODO: Add at least one benchmark to every package type to remove this
+        if: matrix.package == 'core'
+        uses: rhysd/github-action-benchmark@v1
+        with:
+          name: OpenTelemetry Python Benchmarks - Python ${{ env[matrix.python-version ]}} - ${{ matrix.package-group }}
+          tool: pytest
+          output-file-path: output.json
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          # Alert with a commit comment on possible performance regression
+          alert-threshold: 200%
+          comment-always: true
+          fail-on-alert: true
```

Comment:
@aabmass If we fail the test, and merge it into master, it will **not** publish the commit to `gh-pages` because the test failed.

So the question is, is there a case where we would merge a PR which fails performance tests? If yes, then the maintainers who can merge would just have to flip to flag to `false` for the push to `gh-pages` to succeed and flip the flag to back to `true` later to fail subsequent PRs which hurt performance.

The other option is to not fail. In that case the push to `gh-pages` will always succeed, but the CI will pass even when performance gets bad. (But we will still have a comment that says performance got worse!)

cc @lzchen @codeboten

Reply:
> it will **not** publish the commit to `gh-pages` because the test failed.

Hmm that's kind of weird. Is that just because the publishing would happen after the failure?

I think this check should be merge blocking, but ofc there will be times it has to be merged anyway. A decent solution: use a tag similar to "No Changelog" to derive `fail-on-alert`

https://github.com/open-telemetry/opentelemetry-python/blob/5450e6decfbc82b9fa1d2b159272efa33b112c97/.github/workflows/changelog.yml#L16

Reply:
> Is that just because the publishing would happen after the failure?

I think that's the right way to look at it. As a feature of the action it only pushes if the test succeeded.

> A decent solution: use a tag similar to "No Changelog" to derive fail-on-alert

I like this. As in you have to add a "Performance Regression" label in the commit message (has to be commit message because merging into master isn't a `pull_request` event and we only update `gh-pages` when its a merge into master).

@lzchen What do you think?

Reply:
I think if the tests fail, they should be merge blocking. We are talking about changes into core which should uphold these benchmark tests all the time. If there are specific exceptions (which there should be few), we could work some maintainer magic when those times come. :D

Reply:
Sounds good! This is setup to be merge blocking then 😄